### ISSUE TITLE
Inlining code of the Parser and license

### ISSUE THREAD

**Issue by unkind** [hoaproject/Compiler#94]
Hi,

I'd like to use your library, but I want to automatically inline parent class `\Hoa\Compiler\Llk\Parser` and minimize generated class. So far it looks like
```php
class FooParser extends \Hoa\Compiler\Llk\Parser
{
    public function __construct()
    {
        // ...
    }
}
```
The point is to decouple final product from the compiler.

I wanna ask whether it's acceptable for you. Of course, generated class would have reference to your library, e.g.
```php
/**
 * This parser was autogenerated by merging Foo-related stuff and Hoa Compiler Parser into this one.
 * In other words, all the credits regarding generic Parser and Lexer goes to maintainers of https://github.com/hoaproject/Compiler.
 * But rights on the Foo-related part of the software (describing formal grammar, etc.) belongs to Santa-Claus.
 *
 * Generated on Wednesday, November 7, 2018 (UTC) using Hoa Compiler "3.17.08.08".
 */
final class FooParser
{
    // ...
}
```

Text of your license states that I also have to include the whole. Okay, but does it mean that my MIT-licensed product (for example) becomes product with mixed license?

What do you think?

**Comment by Hywan** [comment]
Hello!

Thanks for reaching us. What's your project, and what license are you using?

**Comment by unkind** [comment]
I realized that for my purposes I need only simple lexer, this component is overkill. Regarding license, I use MIT always.

I think we shouldn't bother, but I'm still curious what happens in this cases. AFAIK, it doesn't violate your version of BSD-licence.
